api/seedreconfig/seedreconfig.go

```diff
@@ -0,0 +1,86 @@
+package seedreconfig
+
+type PEM string
+
+const (
+	SeedReconfigurationVersion = 1
+)
+
+// SeedReconfiguration contains all the information that is required to
+// transform a machine started from a SNO seed image (which contains dummy seed
+// configuration) into a SNO cluster with a desired configuration. During an
+// IBU, this information is taken from the cluster that is being upgraded (the
+// original SNO's LCA writes a file to the seed stateroot to communicate this
+// information). During an IBI, this information is placed in the configuration
+// ISO created by the image-based-install-operator.
+//
+// WARNING: Changes to this struct and its sub-structs should not be made
+// lightly, as it is also used by the image-based-install-operator. Any changes
+// made here will also need to be handled in the image-based-install-operator
+// or be backwards compatible. If you've made a breaking change, you will need
+// to increment the SeedReconfigVersion constant to avoid silent breakage and
+// allow for backwards compatibility code.
+type SeedReconfiguration struct {
+	// The version of the SeedReconfiguration struct format. This is used to detect
+	// breaking changes to the struct.
+	APIVersion int `json:"api_version"`
+
+	// The desired base domain for the cluster. Equivalent to install-config.yaml's baseDomain.
+	// This will replace the base domain of the seed cluster.
+	BaseDomain string `json:"base_domain,omitempty"`
+
+	// The desired cluster name for the cluster. Equivalent to install-config.yaml's clusterName.
+	// This will replace the cluster name of the seed cluster.
+	ClusterName string `json:"cluster_name,omitempty"`
+
+	// The desired cluster-ID. During an IBU, this is the ID of the original
+	// SNO. During an IBI, this can either be empty, in which case a new
+	// cluster-ID will be generated by LCA, or it can be set to the ID of the
+	// new cluster, in case one had to be pre-generated for some reason.
+	ClusterID string `json:"cluster_id,omitempty"`
+
+	// The desired IP address of the SNO node.
+	NodeIP string `json:"node_ip,omitempty"`
+
+	// The container registry used to host the release image of the seed cluster.
+	ReleaseRegistry string `json:"release_registry,omitempty"`
+
+	// The desired hostname of the SNO node.
+	Hostname string `json:"hostname,omitempty"`
+
+	// Whether or not the cluster is using a mirror registry.
+	MirrorRegistryConfigured bool `json:"mirror_registry_configured,omitempty"`
```

Comment:
@omertuc @eranco74 actually this field is not need rather than for precaching in future (mirror part is not yet implemented for IBI precache) , it was meant for seed image to know if it had mirror registry or not. 
I assume we will add another fields for idms and icsps, i would remove this one